frontend/src/app/sign-up/page.tsx

```diff
@@ -50,32 +50,31 @@ export default function Home() {
       return;
     }
 
-    // mock for backend
-    const res = await fetch("https://jsonplaceholder.typicode.com/user", {
-      method: "Post",
-      headers: {
-        Accept: "application/json, text/plain, */*",
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        email: email,
-        password: password,
-      }),
+    await fetch(USER_API_ENDPOINT + "/register", {
+        method: "Post",
+        headers: {
+            Accept: "application/json, text/plain, */*",
+            "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+            email: email,
+            password: password,
+            name: "Username placeholder",           // To be changed in the future
```

Comment:
May I know why is the "name" and "major" a placeholder here?

Reply:
Because the registeration form doesn't include name, major. On user-service side, I define the schema to have these two value as non-null as my reason is that a user tuple should have a name and a course(?). Initially I asked frontend devs to consider adding these fields in the registration form but they said this should be added later after the account registered in the service's database so the placeholder is put here to avoid non-null constraints.

Reply:
I see, so I believe there are 2 workarounds:
1. For simplicity, we just include at least name in the registration form so that your database validation will not complain.
2. We need a way to force the user to rename and give a major before we let them do anything.